README.md

```diff
@@ -1 +1,46 @@
 # rancher-desktop-networking
+Is a simple layer 2 switch that allows connectivity between the VM and the host over `AF_VSOCK` written in Go using [gvisor's](https://github.com/google/gvisor) network stack and [gvisor-tap-vsock](https://github.com/google/gvisor). In addition to piping the ethernet frames, it also provides a configurable DNS, dynamic port forwarding and DHCP.
+
+Rancher Desktop Networking consists of three main components: host-switch, vm-switch, and network-setup. Below is a brief explanation of what each component does.
+
+## host-switch
+For now, it only runs on Windows and acts as a receiving end for all traffic that originated from the WSL VM (designated namespace). It performs a handshake to find the right VM to talk to over `AF_VSOCK`. Once a correct VM is found, it then listens for the incoming traffic from that VM. In additon to this, it can provide a configurable DNS resolver that runs in the user space network along with an API that allows for dynamic port forwarding.
+
+## network-setup
+Its main responsibility is to create a network namespace and start the `vm-switch` subprocess in the newly created network namespace.
+
+## vm-switch
+Once the `vm-switch` process starts in the new namespace, it creates a tap (`eth0`) and a `lo` device. Linux Kernel then forwards all the raw Ethernet frames to the tap device. The tap device forwards the frames over [vsock](https://wiki.qemu.org/Features/VirtioVsock) to the host. The process on the host (host-switch) decapsulates the frames, and since it maintains both internal (vm-switch to host-switch) and external (host-switch to the internet) connections, it uses syscall to connect to the external endpoints.
```

Comment:
Something feels off about this…
> Once the `vm-switch` process starts in the new namespace, it creates a tap (`eth0`) and a `lo` device. ~Linux Kernel~ The kernel then forwards all the raw Ethernet frames to the tap device. The tap device forwards the frames over [vsock](https://wiki.qemu.org/Features/VirtioVsock) to the host. The process on the host (~host-switch~ `host-switch`) decapsulates the frames, and ~since it maintains both internal (vm-switch to host-switch) and external (host-switch to the internet) connections, it uses syscall to connect to the external endpoints~ proxies the connections to external hosts via normal syscalls.

Reply:
I wanted to be more elaborate and explicit on how it works.